### PR TITLE
add api usage log for io

### PR DESCRIPTION
**Pull request by kazhang**
Log IO api usages.
Partially-close: #4955

### CODE REVIEW

**Review by datumbox — COMMENTED** [review]
Thanks for the PR @kazhang.

The added logging looks good. Some of the calls (for example `read_image()`) will generate additional logging as they call other public methods but I don't think this is an issue or something we can avoid.

There is at least one more logging needed at the constructor of `VideoReader`  defined [here](https://github.com/pytorch/vision/blob/main/torchvision/io/__init__.py#L108). 

Finally, does it make sense to consider at this point adding the logging calls on the C++ side so that we can include cases where the Python interface is not used? I think that has been discussed offline at one point.

File: torchvision/csrc/io/image/cpu/decode_jpeg.cpp [hunk 1]
```diff
@@ -70,6 +70,7 @@ static void torch_jpeg_set_source_mgr(
 } // namespace
 
 torch::Tensor decode_jpeg(const torch::Tensor& data, ImageReadMode mode) {
+  C10_LOG_API_USAGE_ONCE("torchvision.io.decode_jpeg");
```
Comment (datumbox) [inline torchvision/csrc/io/image/cpu/decode_jpeg.cpp:73]: Do we want to use the same log message as Python's `decode_jpeg` here? Or we want to have a separate distinct message?

File: torchvision/csrc/io/image/cpu/decode_png.cpp [hunk 2]
```diff
@@ -23,6 +23,7 @@ torch::Tensor decode_png(
     const torch::Tensor& data,
     ImageReadMode mode,
     bool allow_16_bits) {
+  C10_LOG_API_USAGE_ONCE("torchvision.io.decode_png");
```
Comment (datumbox) [inline torchvision/csrc/io/image/cpu/decode_png.cpp:26]: Same here and in all other places where the string on the C++ side matches the one on Python.

File: torchvision/csrc/io/video_reader/video_reader.cpp [hunk 3]
```diff
@@ -653,10 +655,12 @@ torch::List<torch::Tensor> read_video_from_file(
 }
 
 torch::List<torch::Tensor> probe_video_from_memory(torch::Tensor input_video) {
+  C10_LOG_API_USAGE_ONCE("torchvision.io.probe_video_from_memory");
```
Comment (datumbox) [inline torchvision/csrc/io/video_reader/video_reader.cpp:658]: @prabhat00155 Do we want to log this? Is this considered part of public API?

**Review by datumbox — APPROVED** [review]
I have a few questions up for discussion, other than that the changes look good to me.

File: torchvision/csrc/io/video_reader/video_reader.cpp (hunk 3 above)
Comment (prabhat00155) [reply torchvision/csrc/io/video_reader/video_reader.cpp:658]: Yeah, see this: https://github.com/pytorch/vision/blob/01ffb3ae9f9cc3a445bcaabf787251a05e67c7a7/torchvision/csrc/io/video_reader/video_reader.cpp#L667

**Review by prabhat00155 — APPROVED** [review]
Thanks @kazhang!

File: torchvision/csrc/io/image/cpu/decode_jpeg.cpp (hunk 1 above)
Comment (kazhang) [reply torchvision/csrc/io/image/cpu/decode_jpeg.cpp:73]: It depends on whether we're particular interested in events from C++ API calls. Our logging system currently couldn't tell whether it's logged in C++ or Python.

I can add "_cpp" suffix here (e.g. `torchvision.io.decode_jpeg_cpp`) for now, we could always further process in log pipeline.

**Review by datumbox — CHANGES_REQUESTED** [review]
@kazhang It seems that something is wrong with the PR: 78 modified files and 3.1k additions/1.6k deletions.

I haven't checked which commit caused this but perhaps its worth force pushing to undo or starting a new one. Marking as "request changes" to avoid accidental merges while the PR is in this state.

**Comment by kazhang** [comment]
@datumbox I created a new PR: #5038

**Pull request closed without merging**